spec/controllers/v0/sign_in_controller_spec.rb

```diff
@@ -2537,27 +2528,79 @@
           end
         end
       end
+
+      context 'and client_id is not present' do
```

Comment:
I think this spec is tested with the 'when client_id is arbitrary spec below (so it can prob be removed)